Document build and test commands

twitchard opened the pull request:
These are the commands that work on my machine. No idea what works in general.

twitchard committed:
Document build and test commands

chrismanahan commented:
trying to get these working on my machine. what're your versions of `swift` and Xcode?

my xcode is 16.4 and 
```
$ swift --version
swift-driver version: 1.120.5 Apple Swift version 6.1.2 (swiftlang-6.1.2.1.2 clang-1700.0.13.5)
Target: arm64-apple-macosx15.0
```

my errors currently
```
$ swift build --triple arm64-apple-ios --sdk $(xcrun --sdk iphoneos --show-sdk-path)
Building for debugging...
error: couldn't build /Users/hume/src/hume-swift-sdk/.build/arm64-apple-ios/debug/Hume.build/TTSClient.swift.o because of multiple producers: Compiling Swift Module 'Hume' (115 sources), Compiling Swift Module 'Hume' (115 sources)

```

```
$ xcodebuild test -scheme Hume-Package -destination 'platform=iOS Simulator,name=iPhone 16'
....
Testing failed:
	Filename "TTSClient.swift" used twice: '/Users/hume/src/hume-swift-sdk/Sources/Hume/API/TTS/Client/TTSClient.swift' and '/Users/hume/src/hume-swift-sdk/Sources/Hume/API/TTS/Resources/TTS/TTSClient.swift'
	Testing cancelled because the build failed.

** TEST FAILED **


The following build commands failed:
	SwiftDriver Hume normal arm64 com.apple.xcode.tools.swift.compiler (in target 'Hume' from project 'Hume')
	Testing workspace hume-swift-sdk with scheme Hume-Package
(2 failures)
```

twitchard committed:
feat: Add CI and document build/test commands

This commit introduces a new GitHub Actions workflow to build and test the Hume Swift SDK. The workflow includes steps for:
- Checking Swift code formatting
- Building the project for iOS devices
- Running tests on the iOS Simulator
- Linting the podspec

Additionally, this commit updates the `CONTRIBUTING.md` file to document these commands.

twitchard committed:
fix(ci): Use iPhone 15 Pro and list simulators

This commit changes the iOS Simulator used for testing to "iPhone 15 Pro" and adds a step to list available simulators to help with debugging.

twitchard committed:
fix(ci): Specify iOS version for simulator

This commit specifies the iOS version for the simulator in the CI workflow to ensure that the correct simulator is used. This should fix the "Unable to find a device" error.

twitchard committed:
fix(ci): Use iPhone 16 Pro for testing

This commit updates the CI workflow to use the "iPhone 16 Pro" simulator with iOS 26.0, which is available in the GitHub Actions runner. This should fix the "Unable to find a device" error.

twitchard committed:
fix(ci): Create and use a simulator for testing

This commit updates the CI workflow to create and boot an iOS simulator before running tests. This ensures that a simulator is available and avoids the "Unable to find a device" error.

The `CONTRIBUTING.md` file has been updated to reflect the new testing strategy and to provide instructions on how to create a simulator locally.
